Source/JavaScriptCore/lol/LOLJIT.cpp

```diff
@@ -0,0 +1,1612 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "LOLJIT.h"
+
+#if ENABLE(JIT) && USE(JSVALUE64)
+
+#include "BaselineJITPlan.h"
+#include "BaselineJITRegisters.h"
+#include "BytecodeGraph.h"
+#include "BytecodeUseDef.h"
+#include "CodeBlock.h"
+#include "CodeBlockWithJITType.h"
+#include "DFGCapabilities.h"
+#include "JITInlines.h"
+#include "JITLeftShiftGenerator.h"
+#include "JITOperations.h"
+#include "JITSizeStatistics.h"
+#include "JITThunks.h"
+#include "LLIntEntrypoint.h"
+#include "LLIntThunks.h"
+#include "LinkBuffer.h"
+#include "MaxFrameExtentForSlowPathCall.h"
+#include "ModuleProgramCodeBlock.h"
+#include "PCToCodeOriginMap.h"
+#include "ProbeContext.h"
+#include "ProfilerDatabase.h"
+#include "ProgramCodeBlock.h"
+#include "SlowPathCall.h"
+#include "StackAlignment.h"
+#include "ThunkGenerators.h"
+#include "TypeProfilerLog.h"
+#include <wtf/BubbleSort.h>
+#include <wtf/GraphNodeWorklist.h>
+#include <wtf/SequesteredMalloc.h>
+#include <wtf/SimpleStats.h>
+#include <wtf/text/MakeString.h>
+
+WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
+
+namespace JSC::LOL {
+
+namespace LOLJITInternal {
```

Comment:
Nit: since this is inside JSC::LOL, could it just be called Internal, and then you refer to JSC::LOL::Internal?